python/ray/train/xgboost/config.py

```diff
@@ -105,3 +185,18 @@ def on_shutdown(self, worker_group: WorkerGroup, backend_config: XGBoostConfig):
                 f"within {timeout} seconds. "
                 "The process will still be terminated as part of Ray actor cleanup."
             )
+
+
+_xgboost_args: dict = {}
+_xgboost_args_lock = threading.Lock()
+
+
+def _set_xgboost_args(args):
+    with _xgboost_args_lock:
+        global _xgboost_args
```

Comment:
Why we need to use a global variable here?

Reply:
needed to pass message from here backend to `backend_cls.train_loop_context_manager`

Reply:
Yeah, I just realized that. Thanks for the answer.